common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -119,7 +127,12 @@ public static void putFloat(Object object, long offset, float value) {
   }
 
   public static double getDouble(Object object, long offset) {
-    return _UNSAFE.getDouble(object, offset);
+    if ( null == _doubleBuffer) {
+      return _UNSAFE.getDouble(object, offset);
+    } else {
+      copyMemory(object, offset, _doubleBuffer.getBaseObject(), _doubleBuffer.getBaseOffset(), 8);
+      return _UNSAFE.getDouble(_doubleBuffer.getBaseObject(), _doubleBuffer.getBaseOffset());
```

Comment:
If `getLong()` never causes an alignment error, how about Double.longBitsToDouble(_UNSAFE.getLong()).  
While I am not familiar with ARM architecture, which one is faster?